static/app/views/issueDetails/streamline/issueUptimeCheckTimeline.tsx

```diff
@@ -0,0 +1,134 @@
+import {useRef} from 'react';
+import {useTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+
+import {CheckInPlaceholder} from 'sentry/components/checkInTimeline/checkInPlaceholder';
+import {CheckInTimeline} from 'sentry/components/checkInTimeline/checkInTimeline';
+import {
+  GridLineLabels,
+  GridLineOverlay,
+} from 'sentry/components/checkInTimeline/gridLines';
+import {useTimeWindowConfig} from 'sentry/components/checkInTimeline/hooks/useTimeWindowConfig';
+import {getTickStyle} from 'sentry/components/checkInTimeline/utils/getTickStyle';
+import {Flex} from 'sentry/components/container/flex';
+import {space} from 'sentry/styles/space';
+import {useDebouncedValue} from 'sentry/utils/useDebouncedValue';
+import {useDimensions} from 'sentry/utils/useDimensions';
+import {
+  checkStatusPrecedent,
+  statusToText,
+  tickStyle,
+} from 'sentry/views/insights/uptime/timelineConfig';
+import {useUptimeMonitorStats} from 'sentry/views/insights/uptime/utils/useUptimeMonitorStats';
+import {useIssueDetails} from 'sentry/views/issueDetails/streamline/context';
+
+export function IssueUptimeCheckTimeline() {
+  const theme = useTheme();
+  const {detectorDetails} = useIssueDetails();
+  const {detectorId, detectorType} = detectorDetails;
+  const elementRef = useRef<HTMLDivElement>(null);
+  const {width: containerWidth} = useDimensions<HTMLDivElement>({elementRef});
+  const timelineWidth = useDebouncedValue(containerWidth, 500);
+  const timeWindowConfig = useTimeWindowConfig({timelineWidth});
+
+  const {data: uptimeStats, isPending} = useUptimeMonitorStats({
+    ruleIds: detectorType === 'uptime_monitor' && detectorId ? [detectorId] : [],
+    timeWindowConfig,
+  });
+
+  return (
+    <ChartContainer>
+      <TimelineGridLineOverlay
+        stickyCursor
+        allowZoom
+        showCursor
+        timeWindowConfig={timeWindowConfig}
+        gridLineStyles={{
+          border: 'none',
+          height: space(0.5),
+          width: 1,
+          borderRadius: 1,
+          background: theme.translucentBorder,
+          top: 68,
+        }}
+        underscanStyles={{
+          margin: 0,
+          height: '100%',
+        }}
+      />
+      <TimelineLegend ref={elementRef} gap={space(1)}>
+        {checkStatusPrecedent.map(status => (
+          <Flex align="center" gap={space(0.5)} key={status}>
```

Comment:
I might have actually just made this `TimelineLegend` and then you can put the text styles in there and just not have the text itself in a container

Reply:
Er sorry `TimelineLegendItem`

Reply:
The text is a little shorter here and we don't want tooltips, so I think we can just have the components be separate for now.

Reply:
👍